src/meds_torch/tune.py

```diff
@@ -81,31 +102,45 @@ def get_checkpoint_path(log_dir, checkpoint_dir_name, time_output_path):
         return checkpoint_files[0]
 
 
+def train_func(cfg):
+    if cfg.hparams_search.train_fn == "train":
+        initialize_objects = initialize_train_objects
+    elif cfg.hparams_search.train_fn == "finetune":
+        initialize_objects = initialize_finetune_objects
+    else:
+        raise ValueError(f"Invalid train_fn: {cfg.hparams_search.train_fn}, should be 'train' or 'finetune'")
+    plugins = [RayLightningEnvironment()]
+
+    object_dict = initialize_objects(cfg, plugins=plugins)
+    dm = object_dict["datamodule"]
+    model = object_dict["model"]
+    trainer = object_dict["trainer"]
+
+    trainer = prepare_trainer(trainer)
+    trainer.fit(model, datamodule=dm)
+
+
 @hydra.main(version_base="1.3", config_path=str(config_yaml.parent.resolve()), config_name=config_yaml.stem)
 def main(cfg: DictConfig) -> float | None:
     """Main entry point for training.
 
-    :param cfg: DictConfig configuration composed by Hydra.
-    :return: Optional[float] with optimized metric value.
+    Args:
+        cfg: DictConfig configuration composed by Hydra.
+    Returns:
+        Optional[float] with optimized metric value.
     """
+    os.environ["RAY_memory_monitor_refresh_ms"] = "0"
```

Comment:
This doesn't seem like something you should globally overwrite.

Reply:
Oh [that disables the RAY memory monitor](https://docs.ray.io/en/latest/ray-core/scheduling/ray-oom-prevention.html) which often kills my jobs.

Reply:
I think the best solution is to add an arg:
`os.environ["RAY_memory_monitor_refresh_ms"] = cfg.ray_memory_monitor_refresh_ms`